indigo/management/commands/datamigrationcleancountry.py

```diff
@@ -0,0 +1,212 @@
+                        + str(
+                            delivery_location.get("location_country", {}).get(
+                                "value", ""
+                            )
+                        )
+                    )
+                    country_code = get_country_code_from_old_data(
+                        delivery_location.get("location_name", {}).get("value", ""),
+                        delivery_location.get("location_country", {}).get("value", ""),
+                    )
+                    if country_code:
+                        print("RESULT Will change country to " + country_code)
+                        delivery_location["location_country"]["value"] = country_code
+                        changes = True
+                    else:
+                        print("RESULT will not change")
+
+            if changes:
+                writes.append(
+                    NewEventData(
+                        project.record.type,
+                        project.record,
+                        {"delivery_locations": delivery_locations},
+                        mode=jsondataferret.EVENT_MODE_MERGE,
+                        approved=True,
+                    )
+                )
+
+        if writes:
+            print("Saving")
+            newEvent(
+                writes, user=None, comment="Data Migration - Clean Country",
+            )
+
+
+def get_country_code_from_old_data(location_name, location_country):
+    if location_name is None:
+        location_name = ""
+    if location_country is None:
+        location_country = ""
+    if "Queensland" in location_name or "Queensland" in location_country:
+        return "AU"
+    if "Manchester" in location_name or "Manchester" in location_country:
+        return "GB"
+    if "United Kingdom" in location_name or "United Kingdom" in location_country:
+        return "GB"
+    if "USA" in location_name or "USA" in location_country:
+        return "US"
+    if "France" in location_name or "France" in location_country:
+        return "FR"
+    if "South Africa" in location_name or "South Africa" in location_country:
+        return "ZA"
+    if "Switzerland" in location_name or "Switzerland" in location_country:
+        return "CH"
+    if (
+        "United Arab Emirates" in location_name
+        or "United Arab Emirates" in location_country
+    ):
+        return "AE"
+    if (
+        "Greater Merseyside" in location_name
+        or "Greater Merseyside" in location_country
+    ):
+        return "GB"
+    if "Australia" in location_name or "Australia" in location_country:
+        return "AU"
+    if "Bristol" in location_name or "Bristol" in location_country:
+        return "GB"
+    if "London" in location_name or "London" in location_country:
+        return "GB"
+    if "UK" in location_name or "UK" in location_country:
+        return "GB"
+    if "West Midlands" in location_name or "West Midlands" in location_country:
```

Comment:
Was the old data of a standard format or is case a possible issue here?

Reply:
It was not of standard format - if you mean case as in upper/lower in "West Midlands" then it matches what explicit instance in old data.